src/dmangle.d

```diff
@@ -88,6 +94,9 @@ private immutable char[TMAX] mangleChar =
     //              X   // variadic T t...)
     //              Y   // variadic T t,...)
     //              Z   // not variadic, end of parameters
+    //              #   // Type backward reference
```

Comment:
Yeah, that might work. There are some ambiguities in the current name mangling, though, so I'm not 100% sure we'd need more context.